### DISCUSSION TITLE
牛叉。就是句子开始的时间戳还不够准。

### DISCUSSION THREAD

**Discussion by niubility000** [ZuodaoTech/everyone-can-use-english#961] [Q&A]
居然发现了这个项目，使用了一下，界面超炫酷啊，能够生成单词级的字幕文件。这个太牛了。
目前发现的不足是生成的字幕每句的起止时间有些还不够准（噢，后来试了那个local，非常非常准）！！！复读功能还可以加强。

local那个字幕识别引擎有些英语单词错了，更不能换成whisper medium-en？而且里面有music等提示，能不能去掉呢？

网上能用的免费的复读软件太少了。我前一阵子就给孩子写了个完全免费的练习英语听力的软件，pdj media repeater。https://github.com/niubility000/PDJ-Media-Repeater  是利用现成的.srt文件和mp3, mp4媒体文件来复读。里面还有很多资源的下载链接。Enjoy可以参考我这个软件里的设置项，可以设置复读间隔时间，可以设置复读次数，可以分别设置每遍复读时的速度，可以加生词和词组，高亮显示。还可以调用系统tts朗读译文（在线的找到个免费的，其他的都要注册账号，没法弄）。而且我比较偷懒，做成了服务器版（演示网站：http://pdj.tpddns.cn:1688  test:test，点击黄色图标的.srt文件就能启动复读机功能。）这样用任何手机或电脑的浏览器都能访问了。否则要给windows，linux，mac, android，ios分别做一个，那就太夸张了。

我发现在pdj媒体复读机里使用enjoy用azure生成的字幕，只要在设置里把每句提前500毫秒，每句结束延后300毫秒就完美了，不会出现读半个词的情况了。enjoy里可以加个全文批量调节句首句尾时间戳的功能?
我用过好多复读软件，字幕时间戳最准的是手机app里的轻听英语，单句复读功能非常好用，左右滑动切换上下句，单击马上重播。字幕应该是他们一句一句手动调的，非常耗人力啊。后来我又研究自动生成字幕的软件，包括微软windows自带的clipchamp，whisper，剪映等。发现剪映6.01以后的版本 (b站上有破解版) 断句和时间戳都变得非常准了，给纯文本配时间戳(剪映里叫文稿匹配) 也非常准，只是有5000词的限制。

今天发现了这个Enjoy，生成的字幕基本不差于剪映了。每句起止的时间戳用local很准，但单词偶尔有识别错的。能否改善呢？文字识别是用的EtienneAb3d/WhisperHallu 吗？能不能把里面的whisper换成whisper-small或medium模型，medium模型很耗时，small就很准了。

而且，Enjoy 目前的版本 音频文件有50兆的限制，视频文件没法选（只支持mp3等音频格式？），这些以后是不是会改善？

加油，希望越改越好。这样纯英文音视频也能直接拿来复读了。

**Comment by niubility000** [comment]
另外，这个软件支持离线使用吗? 能在国内使用吗？是免费的吗？我目前在加州度假，不知道回国了还能不能用。上面有很多youtube的视频，国内打开会不会页面一团乱？其实可以用b站的视频代替，b站里好多英文视频。

**Comment by an-lee** [comment]
感谢关注，就你的问题回答一下。

1. 时间戳的准确性还会继续优化；
2. 大文件的加载后面也会优化，目前限制是为了防止软件卡死；
3. 视频是可以添加的，一般视频格式都支持；
4. 目前国内使用确实有障碍，需要科学上网。

**Reply by niubility000** [reply]
嗯嗯。能放视频啊？貌似有点小bug，找不到视频格式文件，列表显示的只有音频格式的。回头我再找找。
另外，我在上面我的演示网站里用三种能准确生成句子时间戳的软件做了个对比。包括剪映6.1.0，微软clipchamp，还有Enjoy local。 
断句最准的是enjoy，就是文字有些不准，还有>>[music]等这种提示。 
文字最准的是clipchamp，但句子分割不准。 
剪映的文字有些也不准，句子分割有些也不准。

**Reply by bluechallenge** [reply]
补充一些我了解的信息：

1. **Enjoy User Guide**: https://1000h.org/enjoy-app/
2. **语音转文本服务**：Enjoy 软件集成了最小的 whisper 模型 **tiny.en**，如果电脑配置较高，可以选用更大的模型以提高语音转文本的准确度。点击 `模型` 按钮，在弹窗中选择相应的模型即可自动下载并选中。
3. **断句**：如果是少量的断句不准确的话，可以通过 "edit region" 来微调。
<img width="553" alt="image" src="https://github.com/user-attachments/assets/1c634cb4-cbdd-4337-9946-548ad21462eb">

4. **视频资源**：目前 Enjoy 将添加视频的大小限制在 100 MB，超过则会提示添加失败。
<img width="1080" alt="image" src="https://github.com/user-attachments/assets/2a786d7e-d7cb-4340-8f20-52430e917df8">

5. **离线使用**：”文本转语音“ 需要调用 OpenAI 或 AzureAI 服务，AI 翻译、查询、分析以及提取关键词汇也是需要调用OpenAI，所以这些需要在线支持。“语音转文本” 如果用本地模型，那么可以离线使用，如果选在OpenAI/AzureAI/Cloudflare，则需要在线。如果是自己把素材都准备好了，那离线做日常的跟读训练是没啥问题。
6. **是否免费**：如果是调用第三方服务是需要付费，例如：OpenAI/AzureAI/Cloudflare TTS/STT，以及微软 Azure 的发音评估功能。
